src/components/sidebar/sidebar.pw.tsx

```diff
@@ -0,0 +1,441 @@
+
+    await bodyElement.press("Tab");
+    const DialogCloseIconButton = getComponent(page, "dialog").locator(
+      CLOSE_ICON_BUTTON
+    );
+
+    await expect(DialogCloseIconButton).toBeFocused();
+  });
+});
+
+test.describe("Accessibility tests for Sidebar component", () => {
+  test("should check accessibility for SidebarComponent example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<SidebarComponent />);
+
+    await checkAccessibility(page);
+  });
+
+  (["left", "right"] as SidebarProps["position"][]).forEach((boolVal) => {
+    test(`should check accessibility when sidebar position is ${boolVal}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<SidebarComponent position={boolVal} />);
+
+      await checkAccessibility(page);
+    });
+  });
+
+  test("should check accessibility when sidebar has header", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<SidebarComponentWithHeading />);
+
+    await checkAccessibility(page);
+  });
+
+  ([
+    SIDEBAR_SIZES[0],
+    SIDEBAR_SIZES[1],
+    SIDEBAR_SIZES[2],
+    SIDEBAR_SIZES[3],
+    SIDEBAR_SIZES[4],
+    SIDEBAR_SIZES[5],
+    SIDEBAR_SIZES[6],
+  ] as SidebarProps["size"][]).forEach((size) => {
+    test(`should check accessibility when sidebar size is ${size}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<SidebarComponent size={size} />);
+
+      await checkAccessibility(page);
+    });
+  });
+});
+
+test.describe("Check background scroll when tabbing", () => {
+  test("tabbing forward through the sidebar and back to the start should not make the background scroll to the bottom", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<SidebarBackgroundScrollTestComponent />);
+
+    await continuePressingTAB(page, 3);
+    const closeIconButtonElement = closeIconButton(page);
+
+    await expect(closeIconButtonElement).toBeFocused();
+    const boxElement = getDataElementByValue(page, "test-box");
+    await expect(boxElement).not.toBeInViewport();
+  });
+
+  test("tabbing backward through the sidebar and back to the start should not make the background scroll to the bottom", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<SidebarBackgroundScrollTestComponent />);
+
+    await continuePressingSHIFTTAB(page, 1);
+    const closeIconButtonElement = closeIconButton(page);
+
+    await expect(closeIconButtonElement).toBeFocused();
+
+    const boxElement = getDataElementByValue(page, "test-box");
+    await expect(boxElement).not.toBeInViewport();
+  });
+
+  test("tabbing forward through the sidebar and other focusable containers back to the start should not make the background scroll to the bottom", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<SidebarBackgroundScrollWithOtherFocusableContainers />);
+
+    await continuePressingTAB(page, 6);
+    const closeIconButtonElement = closeIconButton(page).nth(0);
+
+    await expect(closeIconButtonElement).toBeFocused();
```

Comment:
**question**: strange for the below GithubActions failure, doesn't seem like there is a value "not focused", any idea?

Reply:
No idea on this as all the jobs have passed 🤷🏻‍♂️ 
How odd. I'll rebase with master and push back up just try to run it with a fresh commit.

Reply:
Fixed this btw. It needed the `waitForAnimationEnd()`as it was flaking.